nilearn/datasets/utils.py

```diff
@@ -527,20 +532,13 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
         shutil.move(temp_full_name, full_name)
         dt = time.time() - t0
         if verbose > 0:
-            print('...done. (%i seconds, %i min)' % (dt, dt // 60))
-    except _urllib.error.HTTPError as e:
-        if verbose > 0:
-            print('Error while fetching file %s. Dataset fetching aborted.' %
-                  (file_name))
-        if verbose > 1:
-            print("HTTP Error: %s, %s" % (e, url))
-        raise
-    except _urllib.error.URLError as e:
+            # Complete the reporting hook
+            sys.stderr.write(' ...done. (%i seconds, %i min)\n' % (dt, dt // 60))
+    except (_urllib.error.HTTPError, _urllib.error.URLError) as e:
         if verbose > 0:
-            print('Error while fetching file %s. Dataset fetching aborted.' %
-                  (file_name))
-        if verbose > 1:
-            print("URL Error: %s, %s" % (e, url))
+            e.message = ("Error while fetching file %s; "
```

Comment:
Good point about `verbose > 0`; now that's not necessary.

I decided to simply change `message`, rather than anything more complicated. We know the types of the errors, so we can be reasonably sure this will work.

If you look at the other commit (c59e5b8) I discarded, I actually made a function to do all the heavy lifting, so that it could be used for any exception. I abandoned it because it was going to be complicated--I wrote tests and everything :)

So, hard to answer your question: I'm going for the least-general / most-specific approach here, and it should work for these errors.

Reading I did:
http://www.ianbicking.org/blog/2007/09/re-raising-exceptions.html
https://stackoverflow.com/questions/6062576/adding-information-to-a-python-exception